packages/ag-grid-theme/css/parts/ag-input.css

```diff
@@ -1,46 +1,98 @@
 /* INPUT */
-
-div[class*="ag-theme-salt"] .ag-filter input[class^="ag-"][type="text"],
-div[class*="ag-theme-salt"] .ag-filter input[class^="ag-"][type="number"],
-div[class*="ag-theme-salt"] .ag-column-select input[class^="ag-"][type="text"],
-div[class*="ag-theme-salt"] .ag-column-select input[class^="ag-"][type="number"] {
+.ag-theme-salt-light .ag-filter input[class^="ag-"][type="text"],
+.ag-theme-salt-dark .ag-filter input[class^="ag-"][type="text"],
+.ag-theme-salt-compact-light .ag-filter input[class^="ag-"][type="text"],
+.ag-theme-salt-compact-dark .ag-filter input[class^="ag-"][type="text"],
+.ag-theme-salt-light .ag-filter input[class^="ag-"][type="number"],
+.ag-theme-salt-dark .ag-filter input[class^="ag-"][type="number"],
+.ag-theme-salt-compact-light .ag-filter input[class^="ag-"][type="number"],
+.ag-theme-salt-compact-dark .ag-filter input[class^="ag-"][type="number"],
+.ag-theme-salt-light .ag-column-select input[class^="ag-"][type="text"],
+.ag-theme-salt-dark .ag-column-select input[class^="ag-"][type="text"],
+.ag-theme-salt-compact-light .ag-column-select input[class^="ag-"][type="text"],
+.ag-theme-salt-compact-dark .ag-column-select input[class^="ag-"][type="text"],
+.ag-theme-salt-light .ag-column-select input[class^="ag-"][type="number"],
+.ag-theme-salt-dark .ag-column-select input[class^="ag-"][type="number"],
+.ag-theme-salt-compact-light .ag-column-select input[class^="ag-"][type="number"],
+.ag-theme-salt-compact-dark .ag-column-select input[class^="ag-"][type="number"] {
   border: none;
   border-bottom: var(--salt-size-border) var(--salt-editable-borderStyle) var(--salt-editable-borderColor);
   height: calc(var(--salt-size-base) + var(--salt-spacing-100));
   padding: 0 var(--salt-spacing-100);
 }
 
-div[class*="ag-theme-salt"] .ag-filter input[class^="ag-"][type="text"]::placeholder,
-div[class*="ag-theme-salt"] .ag-filter input[class^="ag-"][type="number"]::placeholder,
-div[class*="ag-theme-salt"] .ag-column-select input[class^="ag-"][type="text"]::placeholder,
-div[class*="ag-theme-salt"] .ag-column-select input[class^="ag-"][type="number"]::placeholder {
+.ag-theme-salt-light .ag-filter input[type="text"]::placeholder,
```

Comment:
Removed `[class^="ag-"]` which slows down performance, `::placeholder` is scoped well within filter and column select classes